app/api/v1/endpoints/api_key_endpoints.py

```
@@ -1,67 +1,98 @@
+            
+            # Check if IP is in any allowed network
+            if not any(ip in network for network in allowed_networks):
+                raise HTTPException(
+                    status_code=status.HTTP_403_FORBIDDEN,
+                    detail="Access denied: external IP not allowed for internal endpoint"
+                )
+        except ValueError:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="Invalid IP address format"
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)